File: eecs268/2020summer/video-lectures/2020.06.15.interfaces/Triangle.cpp
//Triangle.cpp

#include "Triangle.h"

Triangle::Triangle(double base, double height)
{
	m_base = base;
	m_height = height;
}

double Triangle::area() const
{
	//area = (1/2) * base * height
	return(0.5*m_base*m_height);
}

File: eecs168/2024fall/TR/2024.11.21.classes-magic-methods/trangledemo/triangle.py
#triangle.py

class Triangle:
    def __init__(self, base, height):
        print('Triangle init called')
        self.base = base
        self.height = height

    def area(self):
        ans = (1/2)*self.base*self.height
        return ans
    #           lhs   rhs
    def __gt__(self, other):
        if self.area() > other.area():
            return True
        else:
            return False

    def __eq__(self, other):
        if self.base == other.base and self.height == other.height:
            return True
        else:
            return False

    def __str__(self):
        output_str = 'Triangle with base of '
        output_str += f'{self.base} and '
        output_str += f'height of {self.height}'
        return output_str

    def __repr__(self):
        output_str = f'Triangle({self.base}, {self.height})'
        return output_str
